pkg/vsphere/rbac/rbac.go

```diff
@@ -163,6 +169,65 @@ func (am *AuthzManager) PrincipalBelongsToGroup(ctx context.Context, group strin
 	return false, nil
 }
 
+// PrincipalHasReadOnlyPrivs returns true if the user (principal) in the AuthzManager has
+// at least a read-only role on the input ref object, false otherwise.
+func (am *AuthzManager) PrincipalHasReadOnlyPrivs(ctx context.Context, ref types.ManagedObjectReference) (bool, error) {
```

Comment:
Wil think of a shorter func name for this.

Reply:
Assume we are _always_ checking on a Principal...so could `ReadOnly` or `ReadyOnlyPrincipal`, `ROPrincipal` ... but sure whatever u decide will be kosher

Reply:
IsPrincipalRO?

Reply:
Changed to `ReadOnlyOnRef` as it conveys the intent well. Also we're always checking the `Principal` user as @cgtexmex said, so I removed it from the name.